[CUDA] addmv for complex tensors

anjali411 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#37940 [CUDA] addmv for complex tensors**

Differential Revision: [D21449462](https://our.internmc.facebook.com/intern/diff/D21449462)

anjali411 committed:
[CUDA] addmv for complex tensors

[ghstack-poisoned]

anjali411 committed:
Update on "[CUDA] addmv for complex tensors"




[ghstack-poisoned]

zasdfgbnm reviewed (commented):
Could you also add reinterpret castable test to `cuComplex` and `cuComplexDouble` in https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/test/cuda_complex_test.cu#L36

anjali411 commented:
> Could you also add reinterpret castable test to `cuComplex` and `cuComplexDouble` in https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/test/cuda_complex_test.cu#L36

yeah good idea! will do

anjali411 committed:
Update on "[CUDA] addmv for complex tensors"




[ghstack-poisoned]

anjali411 committed:
Update on "[CUDA] addmv for complex tensors"




[ghstack-poisoned]

anjali411 committed:
Update on "[CUDA] addmv for complex tensors"




[ghstack-poisoned]

anjali411 committed:
Update on "[CUDA] addmv for complex tensors"


Differential Revision: [D21449462](https://our.internmc.facebook.com/intern/diff/D21449462)

[ghstack-poisoned]

anjali411 committed:
Update on "[CUDA] addmv for complex tensors"


Differential Revision: [D21449462](https://our.internmc.facebook.com/intern/diff/D21449462)

[ghstack-poisoned]

anjali411 committed:
Update on "[CUDA] addmv for complex tensors"


Differential Revision: [D21449462](https://our.internmc.facebook.com/intern/diff/D21449462)

[ghstack-poisoned]

zasdfgbnm commented on aten/src/ATen/Dispatch.h:
```diff
@@ -580,6 +606,38 @@ inline void deprecated_AT_DISPATCH_ALL_TYPES_AND_HALF_AND_COMPLEX() {}
 
 #define AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND3(                             \
     SCALARTYPE1, SCALARTYPE2, SCALARTYPE3, TYPE, NAME, ...)                 \
+  [&] {                                                                     \
```
Why this is changed? I guess there is something wrong with the merge conflict. `AT_DISPATCH_ALL_TYPES_AND_C10_COMPLEX_AND3` has been removed and `AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND3` now uses `c10::complex`

anjali411 replied:
yeah fixed it

anjali411 committed:
Update on "[CUDA] addmv for complex tensors"


Differential Revision: [D21449462](https://our.internmc.facebook.com/intern/diff/D21449462)

[ghstack-poisoned]

anjali411 committed:
Update on "[CUDA] addmv for complex tensors"


Differential Revision: [D21449462](https://our.internmc.facebook.com/intern/diff/D21449462)

[ghstack-poisoned]

anjali411 committed:
Update on "[CUDA] addmv for complex tensors"


Differential Revision: [D21449462](https://our.internmc.facebook.com/intern/diff/D21449462)

[ghstack-poisoned]
